### PR TITLE
Add ocnanalbmat

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
**Description**

This adds the ocnanalbmat task to S2S workflow between ocnanalprep and ocnanalrun.

At present this is a stub task that does not actually run the associated jjob, pending more extensive testing

Partially addresses issue https://github.com/NOAA-EMC/GDASApp/issues/263

Files changed:
env/HERA.env
env/ORION.env
parm/config/config.resources
workflow/applications.py
workflow/rocoto/workflow_tasks.py

Files added:
jobs/rocoto/ocnanalbmat.sh
parm/config/config.ocnanalbmat

**Type of change**

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

Tested with abbreviated cycled ocean analysis run on Hera

  
**Checklist**

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
@guillaumevernieres FYI

File: workflow/applications.py [hunk 1]
```diff
@@ -342,13 +346,20 @@ def _get_cycled_task_names(self):
 
         gdas_gfs_common_tasks_before_fcst = ['prep']
         gdas_gfs_common_tasks_after_fcst = ['post', 'vrfy']
+        if self.do_ocean:
+            gdas_gfs_common_tasks_after_fcst += ['ocnpost']
+        gdas_gfs_common_tasks_after_fcst += ['vrfy']
```
Comment (guillaumevernieres) [inline workflow/applications.py]: remove L349-351

**Review by guillaumevernieres — COMMENTED** [review]
looks good. Just a minor comment about removing your edit for `ocnpost` .

**Review by guillaumevernieres — APPROVED** [review]
:+1:

**Comment by AndrewEichmann-NOAA** [comment]
@guillaumevernieres Who in charge of g-w should review this?

File: workflow/rocoto/workflow_tasks.py [hunk 2]
```diff
@@ -516,13 +516,29 @@ def ocnanalprep(self):
...
+                              cdump=self.cdump,
+                              envar=self.envars,
+                              dependency=dependencies)
+
+        return task
+
+    def ocnanalrun(self):
+
+        deps = []
+        dep_dict = {'type': 'task', 'name': f'{self.cdump}ocnanalbmat'}
```
Comment (WalterKolczynski-NOAA) [inline workflow/rocoto/workflow_tasks.py:538]: I think these two task names are reversed? `ocnanalprep` is under `ocnanalbmat` and `ocnanalbmat` is under `ocnanalrun`

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Please add an entry to `WCOSS2.env` in addition to HERA and ORION.

File: workflow/rocoto/workflow_tasks.py (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply workflow/rocoto/workflow_tasks.py:538]: These are the dependencies - `ocnanalbmat` depends on `ocnanalprep`, and `ocnanalrun` on `ocnanalbmat`

Comment (AndrewEichmann-NOAA) [reply workflow/rocoto/workflow_tasks.py:538]: @guillaumevernieres Does any of this run on WCOSS?

Comment (guillaumevernieres) [reply workflow/rocoto/workflow_tasks.py:538]: > @guillaumevernieres Does any of this run on WCOSS?

@AndrewEichmann-NOAA , I don't think we can build the JEDI system yet on WCOSS, so No.

Comment (WalterKolczynski-NOAA) [reply workflow/rocoto/workflow_tasks.py:538]: Oh, duh

Comment (aerorahul) [reply workflow/rocoto/workflow_tasks.py:538]: @guillaumevernieres Technically you can build on WCOSS2.  @CoryMartin-NOAA knows how.

File: parm/config/config.resources [hunk 3]
```diff
@@ -249,6 +249,27 @@ elif [[ "${step}" = "ocnanalprep" ]]; then
...
+        ;;
+      *)
+          echo "FATAL: Resolution not supported'"
+          exit 1
+    esac
+
+    export wtime_ocnanalbmat="00:30:00"
+    export npe_ocnanalbmat=${npes}
+    export nth_ocnanalbmat=1
+    export native_ocnanalbmat="--exclusive"
```
Comment (aerorahul) [inline parm/config/config.resources]: Are we sure this `--exclusive` will work on all schedulers?
I think you want this line to be:
```suggestion
    export is_exclusive=True
```

Please also update the section in `ocnanalrun` where this slipped in.

**Review by aerorahul — CHANGES_REQUESTED** [review]
The use of `exclusive` is incorrect.

File: parm/config/config.resources (hunk 3 above)
Comment (CoryMartin-NOAA) [reply parm/config/config.resources]: not true, a required dependency of IODA is still not available, but OOPS can compile

Comment (AndrewEichmann-NOAA) [reply parm/config/config.resources]: Is `is_exclusive=True` a way of generalizing `sbatch --exclusive`?

Comment (aerorahul) [reply parm/config/config.resources]: That is the intention, yes.

File: workflow/rocoto/workflow_tasks.py (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply workflow/rocoto/workflow_tasks.py:538]: @aerorahul @WalterKolczynski-NOAA Shouldn't a WCOSS2 port be a separate PR?

Comment (aerorahul) [reply workflow/rocoto/workflow_tasks.py:538]: yes.  I don't think we are asking about WCOSS2.

**Review by aerorahul — APPROVED** [review]
looks good now.
Thanks.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by AndrewEichmann-NOAA** [comment]
Thank you!

**Pull request merged**